### PR TITLE
Topbar improvement - changes for search component.

### PR DESCRIPTION
**Pull request by burakukula**
top-bar-search-component
top-bar-search

### CODE REVIEW

File: src/components/buttons/_icon-as-button.scss [hunk 1]
```diff
@@ -145,6 +146,16 @@ $includeHtml: false !default;
         }
       }
     }
+    &--transparent {
```
Comment (aju) [inline src/components/buttons/_icon-as-button.scss]: do the same as `--action`

Comment (burakukula) [inline src/components/buttons/_icon-as-button.scss]: @aju done

File: src/components/header/_header.scss [hunk 2]
```diff
@@ -1,24 +1,15 @@
-$headerDominantColor: #65a6a8;
 $headerHeight:  rhythm(2.5); // 60px;
 
 $includeHtml: false !default;
 
 @if ($includeHtml) {
 
   .sg-header {
-    @include image2x($sgImagesPath + 'header_background@2x.jpg', auto, $headerHeight);
```
Comment (burakukula) [inline src/components/header/_header.scss]: @aju done

File: src/components/search/_search.scss [hunk 3]
```diff
@@ -17,9 +17,41 @@ $includeHtml: false !default;
...
 
+    &__icon {
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      left: gutter(0.25);
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      width: 36px;
```
Comment (aju) [inline src/components/search/_search.scss]: gutter(1.5);

File: scripts/tasks/watch-docs-templates.js [hunk 4]
```diff
@@ -1,9 +1,10 @@
 module.exports = function (gulp, plugins, consts) {
     return function (done) {
         var docsSources = plugins.path.join(consts.DOCS, '**', '*.{html,yml}');
+        var componentsTemplatesSources = plugins.path.join(consts.COMPONENTS, '**', '*.html');
```
Comment (burakukula) [inline scripts/tasks/watch-docs-templates.js]: @kdzwinel `watch` was not watching changes in tempates in components folder.

File: src/components/form-elements/_text-input.scss [hunk 5]
```diff
@@ -43,6 +47,16 @@ $includeHtml: false !default;
       position: relative;
     }
 
+    &--light {
```
Comment (aju) [inline src/components/form-elements/_text-input.scss]: add demo to docs

File: src/components/search/search.html [hunk 6]
```diff
@@ -61,3 +61,25 @@ <h3 class="docs-block__header">Small</h3>
...
+                            <svg class="sg-icon sg-icon--x18">
+                                <use xlink:href="#icon-search"></use>
+                            </svg>
+                        </div>
+                    </button>
+                </div>
+            </div>
+        </div>
+    </div>
+</section>
```
Comment (kdzwinel) [inline src/components/search/search.html]: please add additional empty line at the end of the file - https://github.com/brainly/frontend-tools-configs/blob/master/.editorconfig#L5

**Pull request merged**